src/hotspot/share/gc/g1/g1GCPhaseTimes.cpp

```diff
@@ -481,13 +481,13 @@ double G1GCPhaseTimes::print_post_evacuate_collection_set() const {
   debug_phase(_gc_par_phases[MergePSS], 1);
   debug_phase(_gc_par_phases[ClearCardTable], 1);
   debug_phase(_gc_par_phases[RecalculateUsed], 1);
-  if (G1CollectedHeap::heap()->evacuation_failed()) {
+  if (evacuation_failed) {
     debug_phase(_gc_par_phases[RemoveSelfForwardingPtr], 1);
   }
 
   trace_phase(_gc_par_phases[RedirtyCards]);
   debug_time("Post Evacuate Cleanup 2", _cur_post_evacuate_cleanup_2_time_ms);
-  if (G1CollectedHeap::heap()->evacuation_failed()) {
+  if (_gc_par_phases[RestorePreservedMarks]->uninitialized()) {
```

Comment:
This looks a bit strange, why isn't this check using the passed in `evacuation_failed` like above? And if it can't shouldn't the check be `!_gc_par_phases[RestorePreservedMarks]->uninitialized()`?

Reply:
Probably some oversight when implementing some variants in that area. Will fix.